api-client/src/main/java/com/xing/api/RealCallSpec.java

```diff
@@ -231,9 +237,23 @@ public void cancel() {
         return this;
     }
 
+    /** Timeouts in seconds >0. */
+    @Override public CallSpec<RT, ET> timeouts(int connectTimeout, int readTimeout) {
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+        return this;
+    }
+
     /** Returns a raw {@link Call} pre-building the targeted request. */
     private Call createRawCall() {
-        return api.client().newCall(builder.request());
+        OkHttpClient client = api.client();
+        if (readTimeout > 0 && connectTimeout > 0) {
+            client = client.newBuilder()
```

Comment:
When you do this, you are creating another OkHttpClient which has different threadPool than the one in XingApiClient's instance, this might create threading issues. I guess this is the main reason also that OkHttpClient doesn't allow after creation.